### PR TITLE
Feat/23 auth page verify code page

### LINKED ISSUE #23: [FEAT] Authentication Page -> Verify code page
### Description
Implement the verify code page as shown in the design

### Acceptance Criteria
- Responsive across all screen sizes (desktop, tablet and mobile)
- Write tests for your code.
- Use components where necessary (header, footer, buttons, modal and so on)

### Timeframe
24 hours

[Link to Design](https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=2-10859&t=p2gECcZQmYqvvrn7-4)

### Preview
image

### LINKED ISSUE #58: [FEAT] AUTHENTICATION PAGE - OTP / VERIFICATION CODE input component
### Description

_Implement the Inputs that are responsible for accepting OTP code or Verification code according to the design. The inputs should be dynamic and will be responsible for inputing OTP and verification codes._

### Acceptance Criteria

- The inputs have to be done according to the design.
- The inputs should be responsive and not be disrupted on different screens
- Each input should have an onchange function
- The inputs component should return an appended data of all the individual inputs e.g (148372)
- Each input in the component should be able to recieve only one character 
- The next input should automatically be focused when the a character has been typed in the previous input

### Requirements

The following props should be available for the component

1. Normal input props
2. A dynamic number of inputs to display
3. An input data prop to get the inputs data on any input change

### Testing

1. Ensure inputs are displayed properly according to design and requirements
2. Ensure user cannot type in multiple characters in one input
3. Ensure the correct number of inputs are displayed according to what is provided in the props
5. Ensure the appended individual input data is returned on any input change

### Tasks

- [ ] Create the input component
- [ ] Design an input that will be mapped by the number of inputs props. Each input should include the hover and active states
- [ ] Write an onchange function that takes the input data, appends it with other inputs data, stores the appended data in a state and returns the stored data
- [ ] Write unit tests to ensure that the component works as expected
- [ ] Test the component for responsiveness and functionality

### Timeframe

**This task must be completed on or before Friday 11:59 pm as other components depends on this component** 

### Images
Screenshot 2024-07-17 145544
Screenshot 2024-07-17 145603

### Links
Figma Link
[https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=1-18132&m=dev](url)

### PR DESCRIPTION
**Pull request by Vixs101**
# Description

Implement the verify code page as shown in the design

https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=2-10859&t=p2gECcZQmYqvvrn7-4

Closes #23 

# Changes proposed

## What were you told to do?

[FEAT] Authentication Page -> Verify code page

## What did you do?

* Created of an input-otp component for the page which closes #58. (it was reassigned to me a mentor [TCM] after the assigned developer wasn't able to make the requested changes on time).
* Created a the route for the verify-otp page as suggested in the ROUTING.md file
* implemented the verify-otp page 

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/Videos

Screenshot from 2024-07-21 16-50-26
Screenshot from 2024-07-21 18-45-02

### CODE REVIEW

File: src/app/(admin)/admin/page.test.tsx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { render } from "../../../test/utils";
+import { render } from "~/test/utils";
```
Comment (oreoluwa212) [inline src/app/(admin)/admin/page.test.tsx:1]: why did you work on admin?

**Review by oreoluwa212 — CHANGES_REQUESTED** [review]
Refer back to your task and impement just the OTP component at [#23](https://github.com/hngprojects/hng_boilerplate_nextjs/issues/23)

A lot of things here are not required of you.

NB. you are not to make changes to the global files
constant.js??it's a next project

**Comment by oreoluwa212** [comment]
Make sure to add the already made components(header, footer layouts). p.s attach a video of your working page

Also attach screenshots of test(pnpm test failed)

**Comment by Vixs101** [comment]
I wasn't the one that added the constant.js file, the came when I pulled from the dev branch and also some of the test I pulled from dev are failing too. As for the input component, it's not ready yet that's why I had to improvise and create one. Do I just submit it without the component?

File: src/app/(admin)/admin/page.test.tsx (hunk 1 above)
Comment (Vixs101) [reply src/app/(admin)/admin/page.test.tsx:1]: i didn't work on that , i thing it was lint:fix that did that. The last merge u did before i pulled had lint and test issues but now they are all pinned on me.  while the only thing i did was what was asked of me

**Pull request closed without merging**